src/radar_control.py

```diff
@@ -681,6 +718,76 @@ def run_cfs_scan(
     return freq_data
 
 
+def cfs_block(ave_params, cfs_params_dict, cfs_sockets):
```

Comment:
I am in favour of a slight refactor, where instead of nesting the main logic, you first get the error or early-exit conditions out of the way. So this function would then read:

```python
def cfs_block(ave_params, cfs_params_dict, cfs_sockets):
    aveperiod = ave_params.aveperiod
    if not (
        cfs_params_dict[aveperiod].last_cfs_set_time[aveperiod.beam_iter]
        < time.time() - ave_params.aveperiod.cfs_stable_time
        or aveperiod.cfs_always_run
    ):
        return
    # Then continue with the actual function, but now it isn't nested
```